MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/SpinningSymbolLayerIconActivity.java

```diff
@@ -0,0 +1,180 @@
+      .withLayer(new SymbolLayer(LAYER_ID, SOURCE_ID)
+        .withProperties(PropertyFactory.iconImage(ICON_ID),
+          iconAllowOverlap(true))
+      ), new Style.OnStyleLoaded() {
+        @Override
+        public void onStyleLoaded(@NonNull Style style) {
+          startIconSpinningAnimation();
+        }
+      });
+  }
+
+  /**
+   * Set up and start the icon spinning animation. The Android system ValueAnimator emits a new value, which is
+   * used as the SymbolLayer icons' rotation value.
+   */
+  private void startIconSpinningAnimation() {
+    if (iconSpinningAnimator != null) {
+      iconSpinningAnimator.cancel();
+    }
+    iconSpinningAnimator = ValueAnimator.ofFloat(0, 360);
+    iconSpinningAnimator.setDuration(DESIRED_SECONDS_PER_ONE_FULL_360_SPIN * 1000);
+    iconSpinningAnimator.setInterpolator(new LinearInterpolator());
+    iconSpinningAnimator.setRepeatCount(ValueAnimator.INFINITE);
+    iconSpinningAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+      @Override
+      public void onAnimationUpdate(ValueAnimator valueAnimator) {
+        // Retrieve the new animation number to use as the map camera bearing value
+        Float newIconRotateValue = (Float) valueAnimator.getAnimatedValue();
+        mapboxMap.getStyle(new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+            Layer iconSymbolLayer = style.getLayerAs(LAYER_ID);
+            iconSymbolLayer.setProperties(
+              iconRotate(newIconRotateValue),
+              iconPitchAlignment(ICON_PITCH_ALIGNMENT_MAP)
+            );
+          }
+        });
+      }
+    });
+    iconSpinningAnimator.start();
+  }
+
+  private String loadGeoJsonFromAsset(String filename) {
+    try {
+      // Load GeoJSON file
+      InputStream is = getAssets().open(filename);
+      int size = is.available();
+      byte[] buffer = new byte[size];
+      is.read(buffer);
+      is.close();
+      return new String(buffer, "UTF-8");
+    } catch (Exception exception) {
+      Timber.e("Exception Loading GeoJSON: %s", exception.toString());
+      exception.printStackTrace();
+      return null;
+    }
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    if (iconSpinningAnimator != null) {
+      iconSpinningAnimator.end();
```

Comment:
`cancel()` should be preferred here because `end()` assigns the final value and might send another unnecessary update with that value (I didn't verify this though).

Reply:
Fixed